### ISSUE TITLE
Not receiving VOIP-pushes on ios 13 while app in background or terminated?

### ISSUE THREAD

**Issue by mqaiserbutt** [twilio/voice-quickstart-swift#275]
We are not receiving VOIP-pushes on ios 13 while app in background or terminated (the app is receiving voip-pushes as usual if it's active). On ios < 13 everything is fine. I have implemented this guide [https://github.com/twilio/twilio-voice-ios/blob/Releases/iOS-13-Migration-Guide.md#ios-13--xcode-11](migration-guide) successfully. I have uninstall and reinstall number of times our application as well as recreate the VOIP certificates but no help.

### CALL SID
CAd62660ebd37c0c1c86c42d6566fc3ef8
https://www.dropbox.com/s/hvj6jbh7h8qz591/call.png?dl=0

### Steps to Reproduce
1. Start the app
2. Kill the app or send the app to background
3. Make a call from the other device

#### Expected Behavior
The application starts and we start processing the voip-push.

#### Actual Behavior
The application stays unmoved.

#### Reproduces How Often
100%

#### Logs
https://www.dropbox.com/s/8hrm680dpzq8tin/log.rtf?dl=0

#### Voice iOS SDK
5.0

#### Xcode
11.1

#### iOS Version
IOS 13.1.3

#### iOS Device
iPhone XS Max

**Comment by bchen-twilio** [comment]
Hi @mqaiserbutt 

Thanks for providing the debugging information. From the data I can see using the Call SID you provided, it seems the SDK did successfully register the client with the identity and the push notification was also successfully delivered to the mobile client.

Based on the fact that the app is able to receive and present incoming calls when active but not backgrounded or killed, do you see the app gets terminated by iOS due to not reporting the VoIP push notification to CallKit as incoming call when app receives the `callInviteReceived:` callback?

-bobie

**Comment by mqaiserbutt** [comment]
Yes, yesterday we see the app got terminated by iOS due to not reporting the VoIP push notification to CallKit as incoming call when app received the callInviteReceived: callback. But after that i have many times uninstall and reinstall the app and recreate the certificates as well without any help.

**Comment by bchen-twilio** [comment]
Hi @mqaiserbutt 

With the new PushKit policy introduced in iOS 13, all VoIP push notifications must be reported to CallKit as new incoming calls, **before the program execution goes out of the scope of the `pushRegistry:didReceiveIncomingPushWithPayload:forType:withCompletionHandler:` method**. The Voice SDK returns the call invite synchronously via the `callInviteReceived:` callback method when a Twilio call notification is passed via the `TwilioVoice.handleNotification()` method, thus the application should report the push to CallKit in the callback method. Failure of doing so will result in app termination and worse, iOS will stop giving the VoIP notification to the app if it keeps violating the policy for more than 3 times. Re-install is the only solution when this happens.

Please follow the migration guide and make sure the app always report new incoming calls to CallKit upon receiving VoIP notifications.

Thanks,
-bobie

**Comment by mqaiserbutt** [comment]
We are reporting all calls since yesterday when we faced the crash. Re-installing many times doesn't help any.

**Comment by bchen-twilio** [comment]
@mqaiserbutt does the same problem happen with the Swift quickstart in your environment?

**Comment by mqaiserbutt** [comment]
it works on the sample. Thats very strange. Im checking back integration guide.

**Comment by bchen-twilio** [comment]
@mqaiserbutt Thanks for verifying the issue with the quickstart. Feel free to let us know if the guide needs any update to clear confusions.

Please remember that all VoIP push notifications must be reported to CallKit as new incoming calls, **before the program execution goes out of the scope of the `pushRegistry:didReceiveIncomingPushWithPayload:forType:withCompletionHandler:` method**.

-bobie

**Comment by mqaiserbutt** [comment]
Thanks for the help. We have figured out the issue yesterday and resolved now. Our app was receiving the VOIP pushes in background and terminated both. Just because we didn't report this to callKit the very first notification (due to our own app was crashing when it first got the notification) it stopped sending later notifications. We have handled our app crash and now its getting the notifications as usual. 

Note: Once we are stopped to receive notifications, only way to get back notifications is via reinstalling the app.
